docs/devel/kubemark-guide.md

```diff
@@ -0,0 +1,108 @@
+Kubemark is a performance testing tool which allows users to run experiments on simulated clusters. It’s main use case is scalability testing, as simulated clusters can be much bigger than the real ones. On the other hand it’s not designed for correctness testing, but nethertheless it can help with it, by exposing some problems on master components (API server, controller manager or scheduler) that appear only on bigger clusters (e.g. small memory leaks).
+
+This guide will help you understand what the Kubemark is, what it is not, and how to use it.
+
+## Architecture
+
+On a very high level Kubemark cluster consists of two parts: real master components and a set of “Hollow” Nodes. The prefix “Hollow” means an implementation/instantiation of a component with all “moving” parts mocked out. The best example is HollowKubelet, which pretends to be an ordinary Kubelet, but does not start anything, nor mount any volumes - it just lies it does. More detailed design and implementation details are at the end of this document.
+
+Currently master components run on a dedicated machine(s), and HollowNodes run on an ‘external’ Kubernetes cluster. This design has a slight advantage, over running master components on external cluster, of completely isolating master resources from everything else.
+
+## Requirements
+
+To run Kubemark you need a Kubernetes cluster for running all your HollowNodes and a dedicated machine for a master. Master machine has to be directly routable from HollowNodes. You also need an access to some Docker repository.
+
+Currently scripts are written to be easily usable by GCE, but it should be relatively straightforward to port them to different providers or bare metal.
+
+## Common use cases and helper scripts
+
+Common workflow for Kubemark is:
+- starting a Kubemark cluster (on GCE)
+- running Density tests on Kubemark cluster
+- monitoring test execution and debugging problems
+-turning down Kubemark cluster
+
+In the text there will be comments helpful for anyone who’ll want to port Kubemark to different providers.
+
+### Starting a Kubemark cluster
+
+To start a Kubemark cluster on GCE you need to create an external cluster (it can be GCE or GKE cluster) by yourself and run `test/kubemark/start-kubemark.sh` script. It will do all the setup for you using the configuration stored in `cluster/kubemark/config-default.sh` - you can tweak it however you want, but note that some features may not implemented yet, as implementation of Hollow components/mocks will probably be lagging behind ‘real’ one. For performance tests interesting variables are `NUM_MINIONS` and `MASTER_SIZE`. After start-kubemark script is finished you’ll have a ready Kubemark cluster, script stores a kubeconfig file for talking to the Kubemark cluster in `test/kubemark/kubeconfig.loc`.
+
+Currently single HollowNode uses 50 millicores of CPU and ~60MB or memory, which taking into account default cluster addons and fluentD, allows running ~17.5 HollowNodes per core.
```

Comment:
If it depends on the size, I think that what you should say is that the consume x CPU and Y memory, thus you can fill all the empty space in the cluster setting such limits for them. And then you can say that as an example in N-node cluster with Z-core nodes, this would mean that many HollowNodes.